### DISCUSSION TITLE
Rename BundleInstance to BundleDeployment

### DISCUSSION THREAD

**Discussion by perdasilva** [operator-framework/rukpak#433] [Ideas]
`Instance` is a widely used term in programming and from that perspective, having the entrypoint be a `BundleInstance` create a `Bundle` could be a cause for dissonance and make the mental model harder to understand. Perhaps `BundleDeployment` would make it easier to understand and fits well with the k8s `Deployment` mental model of having a pod spec that creates a pod and a deployment that acts as the top level manager.

**Comment by timflannagan** [comment]
This seems reasonable to me, although I have no strong opinion here. +1 on the change.

**Comment by tylerslaton** [comment]
This would definitely help with the confusion some of the initial stakeholders have mentioned with respect to this API. It also clearly points to existing Kubernetes concepts. I am also +1 to this.

**Comment by exdx** [comment]
This was addressed in #436 -- it's now referred to as a BundleDeployment throughout the codebase
